irr/src/COpenGLDriver.cpp

```diff
@@ -3018,10 +3018,10 @@ IImage *COpenGLDriver::createScreenShot(video::ECOLOR_FORMAT format, video::E_RE
 	if (newImage)
 		pixels = static_cast<u8 *>(newImage->getData());
 	if (pixels) {
-		glReadBuffer(GL_FRONT);
+		if (!Params.Doublebuffer)
+		glReadBuffer(Params.Doublebuffer ? GL_FRONT : GL_BACK);
 		glReadPixels(0, 0, ScreenSize.Width, ScreenSize.Height, fmt, type, pixels);
 		testGLError(__LINE__);
-		glReadBuffer(GL_BACK);
```

Comment:
don't just remove this

Reply:
@appgurueu's argument for removing it is convincing. Why do you think it should be kept?

Reply:
I would rather not change this due to the risk of introducing some obscure bug due to it.

Reply:
I added it back